src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -1139,6 +1140,289 @@ public static IResourceBuilder<T> WithCommand<T>(
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// <para>
+    /// The command will be added to the resource represented by <paramref name="builder"/>.
+    /// </para>
+    /// <para>
+    /// If <paramref name="endpointName"/> is specified, the request will be sent to the endpoint with that name on the resource represented by <paramref name="builder"/>.
+    /// If an endpoint with that name is not found, or the endpoint with that name is not an HTTP endpoint, an exception will be thrown.
+    /// </para>
+    /// <para>
+    /// If no <paramref name="endpointName"/> is specified, the first HTTP endpoint found on the resource will be used.
+    /// HTTP endpoints with an <c>https</c> scheme are preferred over those with an <c>http</c> scheme. If no HTTP endpoint
+    /// is found on the resource, an exception will be thrown.
+    /// </para>
+    /// <para>
+    /// The command will not be enabled until the endpoint is allocated and the resource the endpoint is associated with is healthy.
+    /// </para>
+    /// <para>
+    /// If no <paramref name="method"/> is specified, <c>POST</c> will be used.
+    /// </para>
+    /// <para>
+    /// Specifying a <paramref name="httpClientName"/> will use that named <see cref="HttpClient"/> when sending the request. This allows you to configure the <see cref="HttpClient"/>
+    /// instance with a specific handler or other options using <see cref="HttpClientFactoryServiceCollectionExtensions.AddHttpClient(IServiceCollection, string)"/>.
+    /// If no <paramref name="httpClientName"/> is specified, the default <see cref="HttpClient"/> will be used.
+    /// </para>
+    /// <para>
+    /// The <paramref name="configureRequest"/> callback will be invoked to configure the request before it is sent. This can be used to add headers or a request payload
+    /// before the request is sent.
+    /// </para>
+    /// <para>
+    /// The <paramref name="getCommandResult"/> callback will be invoked after the response is received to determine the result of the command invocation. If this callback
+    /// is not specified, the command will be considered succesful if the response status code is in the 2xx range.
+    /// </para>
+    /// </remarks>
+    public static IResourceBuilder<TResource> WithHttpCommand<TResource>(
+        this IResourceBuilder<TResource> builder,
+        string path,
+        string displayName,
+        [EndpointName] string? endpointName = null,
+        HttpMethod? method = null,
+        string? httpClientName = null,
+        Func<HttpRequestMessage, Task>? configureRequest = null,
+        Func<HttpResponseMessage, Task<ExecuteCommandResult>>? getCommandResult = null,
+        string? commandName = null,
+        string? displayDescription = null,
+        string? confirmationMessage = null,
+        string? iconName = null,
+        IconVariant? iconVariant = null,
+        bool isHighlighted = false)
```

Comment:
This is a big signature. What are the odds we are going to add another parameter to this in the future?

Reply:
It's basically the existing signature for `WithCommand` plus the options specific to HTTP commands. What you suggest as an alternative?